editor/src/components/inspector/inspector.tsx

```diff
@@ -355,73 +357,75 @@ export const Inspector = React.memo<InspectorProps>((props: InspectorProps) => {
           }}
           data-testid={InspectorSectionsContainerTestID}
         >
-          <FlexCol
-            css={{
-              overflowY: 'scroll',
-              width: '100%',
-              height: '100%',
-              position: 'relative',
-              paddingBottom: 50,
-            }}
-          >
-            {rootElementIsSelected ? (
-              <RootElementIndicator />
-            ) : (
-              unless(hideAllSections, <AlignmentButtons numberOfTargets={selectedViews.length} />)
-            )}
-            {unless(
-              hideAllSections,
-              <>
-                {when(isTwindEnabled(), <ClassNameSubsection />)}
-                {anyComponents || multiselectedContract === 'fragment' ? (
-                  <ComponentSection isScene={false} />
-                ) : null}
-              </>,
-            )}
-            <CodeElementSection paths={selectedViews} />
-            <ConditionalSection paths={selectedViews} />
-            {unless(
-              hideAllSections,
-              <>
-                {when(
-                  isMyProject,
-                  <TargetSelectorSection
-                    targets={props.targets}
-                    selectedTargetPath={props.selectedTargetPath}
-                    onSelectTarget={props.onSelectTarget}
-                    onStyleSelectorRename={props.onStyleSelectorRename}
-                    onStyleSelectorDelete={props.onStyleSelectorDelete}
-                    onStyleSelectorInsert={props.onStyleSelectorInsert}
-                  />,
-                )}
-                {when(multiselectedContract === 'fragment', <FragmentSection />)}
-                {unless(
-                  multiselectedContract === 'fragment',
-                  // Position and Sizing sections are shown if Frame or Group is selected
-                  <>
-                    <SimplifiedLayoutSubsection />
-                    <ConstraintsSection />
-                  </>,
-                )}
-                {when(
-                  multiselectedContract === 'frame',
-                  <>
-                    <FlexSection />
-                  </>,
-                )}
-                {when(
-                  multiselectedContract === 'frame' || multiselectedContract === 'wrapper-div',
-                  // All the regular inspector sections are only visible if frames are selected
-                  <>
-                    <StyleSection />
-                    <WarningSubsection />
-                    <ImgSection />
-                    <EventHandlersSection />
-                  </>,
-                )}
-              </>,
-            )}
-          </FlexCol>
+          <DisableControlsInSubtree disable={!canEdit}>
```

Comment:
A small proposal - what if we would have a `PermissionsWrapper` that will be used like this:
```
<PermissionsWrapper permissions={['EDIT']}>
 ... inspector components
</PermissionsWrapper>
```
And inside the permissions wrapper itself it will have the disable logic?
This might be more semantic (even though more high-level). And this way whoever uses it only needs to declare which permissions are interesting, and not have to have to break down the logic themselves.
For example - we might want to have a tooltip for the disabled controllers that says why it is disabled ('you don't have an editing persmission'), and if we use a <PermissionsWrapper> we can make all the changes there instead of every location that is breaking down this disable logic.
What do you think?

Reply:
we did discuss a variant of this in the Discord! I think eventually we may introduce this but only as an abstraction _on top of_ the `DisableControlsInSubtree` – so that the internals of the controls don't need to know about anything permissions. But i'd do any kind of PermissionsWrapper as a follow-up task, once we see how many places we actually end up using `DisableControlsInSubtree`

Reply:
I do like this idea, but agree with Balazs that it should come in a follow up (if we choose to go with it)

Reply:
Oh, definitely. We can use the `<DisableControlsInSubtree>` as a lower-level abstraction and introduce high level wrappers later on.